traffic_ops/app/lib/API/CachegroupFallback.pm

```diff
@@ -0,0 +1,283 @@
+		} else {
+			return $self->alert( "Backup configuration DELETED." );
+		}
+	} else {
+		&log( $self, "No backup Cachegroups found");
+		return $self->not_found();
+	}
+}
+
+sub show {
+	my $self = shift;
+	my $cache_id = $self->param("cacheGroupId");
+	my $fallback_id = $self->param("fallbackId");
+	my $id = $cache_id ? $cache_id : $fallback_id;
+
+	#only integers
+	if ( $id !~ /^\d+?$/ ) {
+		&log( $self, "No such Cachegroup id $id");
+		return $self->not_found();
+	}
+
+	my $cachegroup = $self->db->resultset('Cachegroup')->search( { id => $id } )->single();
+	if ( !defined($cachegroup) ) {
+		&log( $self, "No such Cachegroup $id");
+		return $self->not_found();
+	}
+
+	if ( ($cachegroup->type->name ne "EDGE_LOC") ) {
+		return $self->alert("cachegroup should be type EDGE_LOC.");
+	}
+
+	my $rs_backups = undef;
+
+	if ( defined ($cache_id) && defined ($fallback_id)) {
+		$rs_backups = $self->db->resultset('CachegroupFallback')->search({ primary_cg => $cache_id, backup_cg => $fallback_id}, {order_by => 'set_order'});
+	} elsif ( defined ($cache_id) ) {
+		$rs_backups = $self->db->resultset('CachegroupFallback')->search({ primary_cg => $cache_id}, {order_by => 'set_order'});
+	} elsif ( defined ($fallback_id) ) {
+		$rs_backups = $self->db->resultset('CachegroupFallback')->search({ backup_cg => $fallback_id}, {order_by => 'set_order'});
+	}
+
+	if ( defined ($rs_backups) && ($rs_backups->count > 0) ) {
+		my $response;
+		my $backup_cnt = 0;
+		while ( my $row = $rs_backups->next ) {
+			$response->[$backup_cnt]{"cacheGroupId"} = $row->primary_cg->id;
+			$response->[$backup_cnt]{"cacheGroupName"} = $row->primary_cg->name;
+			$response->[$backup_cnt]{"fallbackName"} = $row->backup_cg->name;
+			$response->[$backup_cnt]{"fallbackId"} = $row->backup_cg->id;
+			$response->[$backup_cnt]{"fallbackOrder"} = $row->set_order;
+			$backup_cnt++;
+		}
+		return $self->success( $response );
+	} else {
+		&log( $self, "No backup Cachegroups");
+		return $self->not_found();
+	}
+}
+
+sub create {
+	my $self = shift;
+	my $cache_id = $self->param('cacheGroupId');
+	my $params = $self->req->json;
+
+	if ( !defined($cache_id)) {
+		my @param_array = @{$params};
+		$cache_id = $param_array[0]{cacheGroupId};
+	}
+
+	if ( !defined($params) ) {
+		return $self->alert("parameters must be in JSON format,  please check!");
+	}
+
+	if ( !&is_oper($self) ) {
+		return $self->forbidden();
+	}
+
+	#only integers
+	if ( $cache_id !~ /^\d+?$/ ) {
+		&log( $self, "No such Cachegroup id $cache_id");
+		return $self->not_found();
+	}
+
+	my $cachegroup = $self->db->resultset('Cachegroup')->search( { id => $cache_id } )->single();
+	if ( !defined($cachegroup) ) {
+		return $self->not_found();
+	}
+
+	if ( ($cachegroup->type->name ne "EDGE_LOC") ) {
+		return $self->alert("cachegroup should be type EDGE_LOC.");
+	}
+
+	my $fallback_config = $self->db->resultset('CachegroupFallback')->find({ primary_cg => $cache_id});
+
+	if ( defined ($fallback_config) ) {
```

Comment:
I don't  understand why this is here? how can i add more fallbacks for a cachegroup? if there is 1 fallback in place, this will always error out.

Reply:
@mitchell852 

Yes. I belive, that is what we have discussed. If one has to update , he has to PUT instead of POST.
POST more than once result in 400 Bad request.

PUT a not POST-ed resource will result in 404 Not found

Reply:
ok